### PR TITLE
[MM-20455] Migrate 'components/integrations/edit_incoming_webhook' module and associated tests to TypeScript 

### PR DESCRIPTION
**Pull request by didithilmy**
#### Summary

Migrates the following files to TypeScript:
- `components/integrations/edit_incoming_webhook/index.ts`
- `components/integrations/edit_incoming_webhook/edit_incoming_webhook.tsx`
- `components/integrations/edit_incoming_webhook/edit_incoming_webhook.test.tsx`

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/15485

#### Related Pull Requests

None

#### Screenshots

None

### CODE REVIEW

**Review by devinbinnie — APPROVED** [review]
LGTM! Thanks @didithilmy!

**Review by larkox — CHANGES_REQUESTED** [review]
LGTM! Thank you very much! Just one small thing about comments.

File: components/integrations/edit_incoming_webhook/edit_incoming_webhook.tsx [hunk 1]
```diff
@@ -1,66 +1,43 @@
...
-            /**
-            * The function to call to get an incoming webhook
-            */
-            getIncomingHook: PropTypes.func.isRequired,
-        }).isRequired,
-    }
-
-    constructor(props) {
+type Props = {
+    team: Team;
```
[the remark this answers is not in the document]
Comment (didithilmy) [reply components/integrations/edit_incoming_webhook/edit_incoming_webhook.tsx:24]: Good point, I think I forgot to move the comments. Should have been fixed now.

**Review by larkox — APPROVED** [review]
LGTM. Thanks!

**Review by DHaussermann — APPROVED** [review]
Tested and passed
- CRUD tested backstage UI for incoming webhooks
- No functional issues foung or JS console errors seen
- Tested on Chrome and desktop
LGTM!
Thanks @didithilmy for this improvement!

**Comment by hanzei** [comment]
Thanks for the contribution @didithilmy. Looking forward to another more :rocket:

**Pull request merged**